main.cpp

```diff
@@ -71,28 +71,291 @@ write 3 UDTs below that EACH have:
+}
+
+void RadioReceiver::RecieverControlInterface::selectAntennaInput(int antennaInputSelection)
+{
+    std::cout << "Antenna Input has been updated: " << antennaInputSelection << std::endl;
+}
+
+void RadioReceiver::RecieverControlInterface::setVolume(float volume)
+{
+    std::cout << "Volume has been updated: " << volume << std::endl;
+}
+
 /*
  copied UDT 2:
  */
 
+struct RadioTransmitter
+{
+    RadioTransmitter();
+    ~RadioTransmitter();
+
+    float maxOutputLevel, minOutputLevel, antennaImpedance, microphoneImpedance;
+    bool monitorEnabled;  
+
+    void enableVoiceMemoryUnit(bool enableVoiceMemoryUnit);
+    void enableAntennaTunner(bool enableAntennaTunner);
+    void enableDataManagementUnit(bool enableDataManagementUnit);
+
+    struct TransmitterControlInterface 
+    {
+        TransmitterControlInterface();
+        ~TransmitterControlInterface();
+
+        int antennaSelection, vfoSelection;
+        float frequency, microphoneGain;
+        std::string mode;
+
+        void setMode(std::string modeSelection);
+        void txEnabled(bool txEnabled);
+        void enableKeyer(bool keyerEnabled);
+    } transmitterControlInterface;
+};
+
+RadioTransmitter::RadioTransmitter() : maxOutputLevel(100.f), minOutputLevel(5.f), antennaImpedance(50.f), microphoneImpedance(4.f), monitorEnabled(true)
+{
+    std::cout << "RadioTransmitter has been constructed." << std::endl;
+}
+
+RadioTransmitter::~RadioTransmitter()
+{
+    std::cout << "RadioTransmitter has been deconstructed." << std::endl;
+}
+
+RadioTransmitter::TransmitterControlInterface::TransmitterControlInterface() : antennaSelection(1), vfoSelection(2), frequency(7.100f), microphoneGain(2.f), mode("AM")
+{
+    std::cout << "TransmitterControlInterface has been constructed." << std::endl;
+}
+
+RadioTransmitter::TransmitterControlInterface::~TransmitterControlInterface()
+{
+    std::cout << "TransmitterControlInterface has been deconstructed." << std::endl;
+}
+
+void RadioTransmitter::enableVoiceMemoryUnit(bool enableVoiceMemoryUnit)
+{
+       std::cout << "Voice Memory Unit Enabled: " << enableVoiceMemoryUnit << std::endl;
+}
+ 
+void RadioTransmitter::enableAntennaTunner(bool enableAntennaTunner)
+{
+       std::cout << "Antenna Tunner Enabled: " << enableAntennaTunner << std::endl;
+}
+
+void RadioTransmitter::enableDataManagementUnit(bool enableDataManagementUnit)
+{
+       std::cout << "Data Management Unit Enabled: " << enableDataManagementUnit << std::endl;
+}
+
+void RadioTransmitter::TransmitterControlInterface::setMode(std::string modeSelection)
+{
+       std::cout << "Tx Mode has been set to: " << modeSelection << std::endl;
+}
+
+void RadioTransmitter::TransmitterControlInterface::txEnabled(bool txEnabled)
+{
+       std::cout << "Tx Enabled: " << txEnabled << std::endl;
+}
+
+void RadioTransmitter::TransmitterControlInterface::enableKeyer(bool keyerEnabled)
+{
+       std::cout << "Keyer Enabled: " << keyerEnabled << std::endl;
+}
```

Comment:
please fix the indentation differences.
see the Readme.MD for the course coding standard, which defines the indentation settings you should be using.
- [ ] revise